app/src/main/java/com/bps/gotwinter2021/ui/overview/OverviewViewModel.kt

```diff
@@ -5,15 +5,71 @@ import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import com.bps.gotwinter2021.data.model.GOTResponse
+import com.bps.gotwinter2021.favorites.database.Favorite
 import com.bps.gotwinter2021.favorites.database.FavoriteDatabaseDao
+import kotlinx.coroutines.*
 
 class OverviewViewModel(
-    val application: Application, val database: FavoriteDatabaseDao, val passedCharacterSA: GOTResponse
+    val application: Application, val dataSource: FavoriteDatabaseDao, val passedCharacterSA: GOTResponse
 ) : ViewModel() {
+    private val viewModelJob = Job()
+    private val coroutineScope = CoroutineScope(viewModelJob + Dispatchers.Main)
+
     private val _passedCharacter = MutableLiveData<GOTResponse>()
     val passedCharacter: LiveData<GOTResponse>
         get() = _passedCharacter
+
     init{
         _passedCharacter.value = passedCharacterSA
+        initializeFav()
+    }
+
+    private var fav = MutableLiveData<Favorite?>()
+
+    private fun initializeFav() {
+        coroutineScope.launch{
+            fav.value = getFavFromDatabase()
+        }
+    }
+
+    private suspend fun getFavFromDatabase(): Favorite? {
+        return  withContext(Dispatchers.IO){
+            var fav = dataSource.getNewest()
+            fav
+        }
+    }
+
+    fun addClicked(oldResponse: GOTResponse){
+        coroutineScope.launch {
```

Comment:
Juan and I use viewModelScope.launch(dispatcher) instead of coroutineScope.launch